src/main/java/org/elasticsearch/cluster/metadata/SnapshotMetaData.java

```diff
@@ -330,7 +332,16 @@ public Entry snapshot(SnapshotId snapshotId) {
         return null;
     }
 
+    @Override
+    public MetaData.Custom readFrom(StreamInput in) throws IOException {
+        return FACTORY.readFrom(in);
```

Comment:
Please break BWC and forget about 1.6 - we have to get 2.0 out there and we have to fix this without BWC in mind!